edward/inferences/variational_inference.py

```diff
@@ -87,38 +89,36 @@ def initialize(self, optimizer=None, var_list=None, use_prettytensor=False,
 
       self.summarize = tf.summary.merge_all(key=summary_key)
 
-    if optimizer is None:
+    learning_rate = 0.01
+    if optimizer is None or isinstance(global_step, tf.Variable):
       # Use ADAM with a decaying scale factor.
-      global_step = tf.Variable(0, trainable=False, name="global_step")
+      if not global_step:
```

Comment:
cool 💻  back from the 💀 ... anyways, so 1 question, do I remove the option that the  `optimizer` is of type `tf.train.Optimizer`, e.g. from the `TypeError` string and function doc?

Since it can now be either `None` or a string.

Thanks!